src/game.cpp

```diff
@@ -344,7 +352,6 @@ void game::load_data_from_dir(const std::string &path)
 
 game::~game()
 {
-    DynamicDataLoader::get_instance().unload_data();
```

Comment:
I suspect all those unload / reset functions don't need to be called in the destructor at all. Things should clear themself and nothing outside.

`mapbuffer::~mapbuffer` already class its reset method, so it's not needed here.
`DynamicDataLoader::unload_data` shouldn't be needed at all (in any destructor), the game data should be cleared via the destructor of the various containers. It is needed only before loading the game data or after a game has been saved / lost (not related to the program end at all).

`world_generator` and `gamemode` should be `unique_ptr` and thereby deleting themself as needed.